### ISSUE TITLE
row deselection

### ISSUE THREAD

**Issue by bobabarbanel** [olifolkerd/tabulator#702]
When you use the tabulator function like this
**dT.tabulator("deselectRow");** 
all rows are deselected, BUT the defined 
**rowDeselected: function() {}** 
IS NOT called for each row.

These forces me to loop through the currently selected row, and perform the action that is usually handled in the rowDeselected function.

A related problem. This **crashes**:
dT.tabulator("getSelectedRows").forEach( 
            function(row) {
                row.deselect();
            }
        );

**Comment by olifolkerd** [comment]
Hey @bobabarbanel 

Good point, i will update this behaviour in the next release!

Cheers

Oli

**Comment by bobabarbanel** [comment]
I also find when getting all rows that some (starting with 41st for me) return [] with getCells(). 

Bob Abarbanel
+1 206-499-1443
(Pleas forgive 21st speling errors. 😜)

**Comment by olifolkerd** [comment]
Hey @bobabarbanel

you will be happy to hear i have pushed a fix for this to the 3.4 branch, which will be released in the the next couple of weeks, the ***rowSelected*** and ***rowDeselected*** callbacks are now triggered for each row.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @bobabarbanel 

If the row hasn't been drawn yet (if it is far below the visible set of rows and hasn't been scrolled to yet) it wont have been rendered yet and so wont contain any Cell Components.

Cheers

Oli

**Comment by cemmons** [comment]
Oli, I'm using v3.5.1 and the rowDeselected event is not triggering when using $('').tabulator('deselectRow')

**Comment by olifolkerd** [comment]
Hey @cemmons 

Could you post the code you are using to deselect the row?

Cheers

Oli :)

**Comment by helios0x92** [comment]
Hey @olifolkerd 

I'm using v3.5.2 and the ```rowDeselected``` event is only triggered if user manually deselects the row. or if a call ```row.deselect()``` . If I try to deselect all rows at once in the following way, ```rowDeselected``` event is not triggered:
```javascript
$("#data-table").tabulator("deselectRow");
```

This is why i have to use it the same way @bobabarbanel did : 
```javascript
$("#data-table").tabulator("getSelectedRows").forEach(function (row) {
     row.deselect();
 });
```

However, ```rowSelected``` event triggers in both cases. It would be great if the behaviour was consistent.

**Comment by olifolkerd** [comment]
Hey @helios0x92 

i have just pushed a fix for this to the 4.0 branch and will include it in the next release

**Comment by helios0x92** [comment]
Thanks! @olifolkerd . v4.0 looks great :)
